### DISCUSSION TITLE
Isn't there any jsPsych.pluginAPI.compare"Buttons" ?

### DISCUSSION THREAD

**Discussion by mary-9555** [jspsych/jsPsych#2190] [Q&A]
Hi again
I need _feedback_ for the "jspsych-image-button-response" plugin.
As my choices are not "value", I got these errors:

1. You must specify a value for the choices parameter in the image-button-response plugin.
setDefaultValues @ jspsych.js:1144
jspsych-image-button-response.js:166

2. Uncaught TypeError: Cannot read property 'length' of undefined
    at Object.jsPsych.plugins.image-button-response.plugin.trial (jspsych-image-button-response.js:166)
    at doTrial (jspsych.js:1019)
    at nextTrial (jspsych.js:967)
    at Object.window.jsPsych.core.finishTrial (jspsych.js:339)
    at end_trial (jspsych-image-button-response.js:306)
    at after_response (jspsych-image-button-response.js:285)
    at HTMLDivElement.<anonymous> (jspsych-image-button-response.js:254)
p%3E:1 

3. Failed to load resource: net::ERR_FILE_NOT_FOUND

any advice?

**Comment by jodeleeuw** [comment]
Hi @mary-9555,

It would help a lot if you shared the code that generated these errors. It sounds like you may not have a `choices` parameter specified in one of your `image-button-response` trials.

**Reply by mary-9555** [reply]
Sure!
Here it is:

var image_trial = {
  type: 'image-button-response',
  stimulus: 'img/b/004_o_m_s_b.jpeg',
  choices: ['sad', 'frightened', 'neutral','angry','disgusting','happy'],
  data: {
    stimulus_type: 'image'
  },
  on_finish: function(data){
    if(jsPsych.pluginAPI.compareKeys(data.response, "sad")){
      data.correct = true;
    } else {
      data.correct = false; 
    }
  }
}

var feedback = {
  type: 'instructions',
  stimulus: function(){
    var last_trial_correct = jsPsych.data.get().last(1).values()[0].correct;
    if(last_trial_correct){
      return "<p>Correct!</p>"; 
    } else {
      return "<p>Wrong!... The correct answer was 'sad'.</p>";
      show_clickable_nav; true,
    button_label_previous; 'previous',
    button_label_next; 'next'
    }
  }
}

timeline.push(image_trial, feedback);
